c7n/resources/sns.py

```diff
@@ -119,3 +119,93 @@ def process_resource(self, client, resource):
         return {'Name': resource['TopicArn'],
                 'State': 'PolicyRemoved',
                 'Statements': found}
+
+
+@SNS.action_registry.register('modify-statements')
+class ModifyPolicyStatement(ModifyPolicyBase):
+    """Action to modify policy statements from SNS
+
+    :example:
+
+    .. code-block:: yaml
+
+           policies:
+              - name: sns-cross-account
+                resource: sns
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: modify-statements
+                    add-statements: [statement]
+                    remove-statements: [statement] or *
+    """
+
+    permissions = ('sns:SetTopicAttributes', 'sns:GetTopicAttributes')
+
+    def process(self, resources):
+        replace = False
+        results = []
+        client = local_session(self.manager.session_factory).client('sns')
+        additions = self.data.get('add-statements', [])
+        deletions = self.data.get('remove-statements', [])
+
+        if deletions == "*":
+            replace = True
+
+        for r in resources:
+            new_policy = {u'Version': u'2012-10-17', u'Statement': []}
+            try:
+                if replace:
+                    new_policy = self.process_replace(client, r)
+                if len(additions) and len(deletions) and not replace:
+                    new_policy['Statement'] += self.process_add_and_delete(client, r)
+                if len(additions) and not len(deletions) and not replace:
+                    new_policy['Statement'] += self.process_addition(client, r).get('Statement')
+                if not len(additions) and len(deletions) and not replace:
+                    new_policy['Statement'] += self.process_deletion(client, r)
+                results += {
+                    'Name': r['TopicArn'],
+                    'State': 'PolicyModified',
+                    'Statements': new_policy
+                }
+            except Exception:
+                self.log.exception(
+                    "Error processing sns:%s", r['TopicArn'])
+            else:
+                client.set_topic_attributes(
+                    TopicArn=r['TopicArn'],
+                    AttributeName='Policy',
+                    AttributeValue=json.dumps(new_policy)
+                )
+        return results
+
+    def process_add_and_delete(self, client, resource):
+        policy = resource.get('Policy') or '{}'
+        policy = json.loads(policy)
```

Comment:
this deserialization should be hoisted up instead of duplicated in the processor methods.